packages/shared/isValidElementType.js

```diff
@@ -28,28 +28,39 @@ import {
 } from 'shared/ReactSymbols';
 
 export default function isValidElementType(type: mixed) {
-  return (
-    typeof type === 'string' ||
-    typeof type === 'function' ||
-    // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
-    type === REACT_FRAGMENT_TYPE ||
-    type === REACT_PROFILER_TYPE ||
-    type === REACT_DEBUG_TRACING_MODE_TYPE ||
-    type === REACT_STRICT_MODE_TYPE ||
-    type === REACT_SUSPENSE_TYPE ||
-    type === REACT_SUSPENSE_LIST_TYPE ||
-    type === REACT_LEGACY_HIDDEN_TYPE ||
-    (typeof type === 'object' &&
-      type !== null &&
-      (type.$$typeof === REACT_LAZY_TYPE ||
-        type.$$typeof === REACT_MEMO_TYPE ||
-        type.$$typeof === REACT_PROVIDER_TYPE ||
-        type.$$typeof === REACT_CONTEXT_TYPE ||
-        type.$$typeof === REACT_FORWARD_REF_TYPE ||
-        type.$$typeof === REACT_FUNDAMENTAL_TYPE ||
-        type.$$typeof === REACT_RESPONDER_TYPE ||
-        type.$$typeof === REACT_SCOPE_TYPE ||
-        type.$$typeof === REACT_BLOCK_TYPE ||
-        type[(0: any)] === REACT_SERVER_BLOCK_TYPE))
-  );
+  if (typeof type === 'string' || typeof type === 'function') return true;
+
+  if (
+    [
+      REACT_FRAGMENT_TYPE,
+      REACT_PROFILER_TYPE,
+      REACT_DEBUG_TRACING_MODE_TYPE,
+      REACT_STRICT_MODE_TYPE,
+      REACT_SUSPENSE_TYPE,
+      REACT_SUSPENSE_LIST_TYPE,
+      REACT_LEGACY_HIDDEN_TYPE,
+    ].indexOf(type) !== -1
```

Comment:
A switch/case would probably read just as nicely without the overhead of creating an new array and calling `indexOf` each time.

Reply:
Thank you for review, yes I agree with you, I will update it by `switch/case`